plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -58,7 +69,7 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
   private val mapLongClickResolver = MapLongClick()
   private val mapMoveResolver = MapMove()
   private var draggedAnnotation: T? = null
-  private val annotationMap = mutableMapOf<Long, T>()
+  private val annotationMap = ConcurrentHashMap<Long, T>()
```

Comment:
why do we concurrency here?

Reply:
HashMap may be read and modified at the same time and cause `ConcurrentModificationException` error

Reply:
Iterators must be used to handle it properly
https://stackoverflow.com/questions/602636/why-is-a-concurrentmodificationexception-thrown-and-how-to-debug-it

Reply:
We don't have the same logic that removes items while iterating this map. It happens more like one thread is reading while one thread is modifying. Sadly I can't reproduce it anymore, use `ConcurrentHashMap` just to prevent this issue happens again.

Reply:
@Chaoba - I checked impl and it seems it should be accessed from one thread only.
What's the potential situation when we have several threads working with this map? I guess it's pretty important because if we have concurrency in general - it's definitely not enough to just make one map concurrent, there may be other hard to catch bugs.

Reply:
We can't prevent users from invoking different interfaces from outside at the same time unless we change all the interfaces to synchronous. For example, we [create annotations](https://github.com/mapbox/mapbox-maps-android/pull/122/files#diff-7fb27c886fce919b4fd72a2464f1b8f982933f1ec65355410969b4d7b4f8a7e1R96) from the worker thread while we may [delete annotations](https://github.com/mapbox/mapbox-maps-android/pull/122/files#diff-7fb27c886fce919b4fd72a2464f1b8f982933f1ec65355410969b4d7b4f8a7e1R72) from the UI thread.

Reply:
I understand it all but mostly all our SDK is non thread-safe and we need to state it.
Of course we could easily call any our SDK methods from one thread and another method from another thread and most likely get some error.
My main concern here is that most likely you did not call your annotation methods from different threads 😄 Annotation plugin is designed to be non thread-safe so using `ConcurrentHashMap` seems more like a workaround for some issue.

Could that hash map be modified somehow in result callback of `queryRenderFeatures`? It uses another thread to deliver results. As I see - it's the only place where we could have another thread.

Reply:
`queryRenderFeatures` will only be invoked with a gesture, but in the test example, we don't enable any gesture. 
I have reproduced the crash: firstly change it to normal HashMap and then click the deleteAll button quickly while opening the example.
However if change it to ConcurrentHashMap, the crash will not happen anymore. This verified ConcurrentHashMap does prevent such issues and makes the SDK more robust.

Reply:
Let me check it by myself, I'm very curious about it.

Reply:
@Chaoba checked your branch - this issue could be fixed if using iterators everywhere while it makes code a bit more messy or indeed use `ConcurrentHashMap` as you do. 
Let's go your way.
Another thing I mentioned - if open `SymbolClusterActivity` and quickly press delete button until nothing is loaded - all annotations will still appear on screen after they get calculated. I guess it's wrong behaviour.